mne/io/edf/edf.py

```diff
@@ -602,7 +641,33 @@ def _read_edf_header(fname, exclude):
 
         nchan = int(_edf_str(fid.read(4)))
         channels = list(range(nchan))
-        ch_names = [fid.read(16).strip().decode('latin-1') for ch in channels]
+
+        # read in 16 byte labels and strip any extra spaces at the end
+        signal_labels = [fid.read(16).strip().decode('latin-1')
+                         for ch in channels]
+
+        # ch_names = signal_labels.copy()
+        # get channel names and optionally channel type
+        # EDF specification contains 16 bytes that encode channel names,
+        # optionally prefixed by a string representing channel type separated
+        # by a space
+        ch_names = []
+        ch_types = []
+        for ch, new_label in enumerate(signal_labels):
+            # if no channel type, then we will default to eeg
+            ch_type = None
+            ch_name = new_label
+
+            # space is found, so the prefix is the channel type. 'Annotations'
+            # is also a keyword we search for
+            if new_label.count(' ') == 1 and ('Annotations' not in new_label):
+                ch_type, ch_name = new_label.split(' ')
+
+                # channel types should be upper case for easy comparison
+                ch_type = ch_type.upper()
+
+            ch_names.append(ch_name)
+            ch_types.append(ch_type)
```

Comment:
@Teuniz what do you think about this these changes to our EDF writer? thx

Reply:
Just one question, what is going to happen if, an inputfile in EDF format which contains (besides EEG signals) signals  coming from, for example, an 3-axis accelerometro, is going to be exported to EDF?  
In other words, what happens if a signals' type isn't covered by a standard list of types?  
  
Assigning the type EEG (because it's the fallback)  to an accelerometer signal doesn't seem to be a good idea.  
Because it's practically impossible to cover any kind of sensor and future developments, maybe it could be an idea  
 to have a type unknown or unspecified?  
Just brainstorming here...

Reply:
We do have a `'MISC'` type that we could assign in this case.

Reply:
Upon reading, MNE has traditionally assumed EEG by default, so I think assuming it by default is fine, unless we want to deprecate that behavior?

Unless you mean the following:
1. if there **are** channel types encoded in the EDF signal labels, and then only some of them don't have it, then those should be easily assigned `'misc'`?
2. otherwise if there are no channel types at all, then the default backwards-compatible behavior is assuming `'eeg'`

Should I implement 1?
wdyt @agramfort ?

Upon writing... I agree writing the channel type is not always desirable if we want to control for erroneous user behavior. Thus I added the `add_ch_type` kwarg to export per @cbrnr 's suggestion.